backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/ddl/functions/Fresh_Account_Info_Report.sql

```diff
@@ -337,121 +100,481 @@ CREATE OR REPLACE FUNCTION report.fresh_account_info_report(
+
+                          LEFT OUTER JOIN C_Period period_LastYearEnd ON (period_LastYearEnd.C_Period_ID = report.Get_Predecessor_Period_Recursive($3, (SELECT PeriodNo::int FROM C_Period WHERE C_Period_ID = $3 AND isActive = 'Y'))) AND period_LastYearEnd.isActive = 'Y'
+                 WHERE fa.postingtype IN ('A', 'Y') -- task 09804 don't show/sum other than current (A) -- now we have other current Y (year end)
+                ) fa
+                    LEFT OUTER JOIN GL_JournalLine jl ON fa.Line_ID = jl.GL_JournalLine_ID AND fa.AD_Table_ID = (SELECT Get_Table_ID('GL_Journal')) AND jl.isActive = 'Y'
+                    LEFT OUTER JOIN AD_Table tbl ON fa.AD_Table_ID = tbl.AD_Table_ID AND tbl.isActive = 'Y'
+                    LEFT OUTER JOIN C_BPartner bp ON fa.C_BPartner_ID = bp.C_BPartner_ID AND bp.isActive = 'Y'
+                    LEFT OUTER JOIN C_Activity a ON fa.C_Activity_ID = a.C_Activity_ID AND a.isActive = 'Y'
+                    LEFT OUTER JOIN C_Activity pa ON COALESCE(pa.C_Activity_ID = $7, FALSE) AND pa.isActive = 'Y'
+
+           WHERE CASE
+                     WHEN ($8 = 'Y') THEN
+                             1 = 1
+                                     ELSE
+                         (
+                                 fa.DocStatus NOT IN ('CL', 'VO', 'RE')
+                             )
+                 END
+
+             AND fa.ad_org_id = $11
+
+             AND CASE
+                     WHEN ($10 = 'N') THEN
+                             1 = 1
+                                      ELSE
+                         (
+                             fa.C_Activity_ID IS NULL
+                             )
+                 END
+
+           ORDER BY fa.Param_Acct_Value,
+                    fa.Account_ID,
+                    fa.DateAcct::Date,
+                    fa.Fact_Acct_ID;
+    END INIT_TABLE;
+
+    <<UPDATE_TABLE>>
+        DECLARE
+        currencies numeric[];
+        length int;
+        row report.tmp_fresh_Account_Info_Report_Sub%rowtype;
+        v_currencyid1 numeric;
+        v_currencyid2 numeric;
+        v_currencyid3 numeric;
+        v_currencyid4 numeric;
+        v_currencyid5 numeric;
+        v_currency1 text;
+        v_currency2 text;
+        v_currency3 text;
+        v_currency4 text;
+        v_currency5 text;
+        v_sourcebalance1 numeric;
+        v_sourcebalance2 numeric;
+        v_sourcebalance3 numeric;
+        v_sourcebalance4 numeric;
+        v_sourcebalance5 numeric;
+        v_rollingbalance1 numeric;
+        v_rollingbalance2 numeric;
+        v_rollingbalance3 numeric;
+        v_rollingbalance4 numeric;
+        v_rollingbalance5 numeric;
+    BEGIN
+        currencies := ARRAY (SELECT DISTINCT source_currency_id FROM report.tmp_fresh_Account_Info_Report_Sub);
+        length := array_length(currencies, 1);
+
+        IF length > 0 AND length <= 5 AND $9 = 'Y'THEN --p_showcurrencyexchange (used for option show foreign currency)
```

Comment:
IMHO if show currnecy exchange and array_length(currencies, 1) > 5 you shall raise exception

Reply:
also, in case of "$9 = 'Y'" the whole block makes no sense.
even fetching the distinct currencies is not needed

Reply:
This variable is used for display logic of foreign currencies. I didn’t do any fixing for old code parts like this because of limited time

Reply:
In my opinion I shouldn’t raise an exception, because without the report is still printed without this extra part